src/drivers/camera_trigger/camera_trigger.cpp

```diff
@@ -571,14 +586,15 @@ CameraTrigger::Run()
 
 			/*
 			 * TRANSITIONAL SUPPORT ADDED AS OF 11th MAY 2017 (v1.6 RELEASE)
-			*/
+			 */
 
 			if (cmd.param1 > 0.0f) {
 				_distance = cmd.param1;
 				param_set_no_notification(_p_distance, &_distance);
 
 				_trigger_enabled = true;
 				_trigger_paused = false;
+				_valid_position = false;
```

Comment:
It's not hacky because as you re-enable the trigger you do not have a valid position of the last trigger (the last position stored is from the last cycle, and you've flown a turnaround curve since, so the distance is completely meaningless - you're entering a new cycle). So this is in fact correct.